src/FastExpressionCompiler/FastExpressionCompiler.cs

```diff
@@ -1579,11 +1579,7 @@ private static bool TryEmitIndex(IndexExpression expr, ILGenerator il)
 
                 if (expr.Arguments.Count == 1) // one dimensional array
                 {
-                    if (elemType.IsValueType())

```

Comment:
I replaced this block of code with TryEmitArrayIndex method call to reduce code duplication

Reply:
Ok.